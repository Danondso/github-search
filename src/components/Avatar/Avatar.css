.avatar-container {
  display: inline-block;
  align-content: center;
}

.avatar-image {
  height: 80px;
  width: 80px;
  border-radius: 100px;
  background-color: #282c34;
}
