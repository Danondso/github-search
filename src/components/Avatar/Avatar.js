import React from 'react';
import PropTypes from 'prop-types';
import './Avatar.css';

function Avatar({ imageSrc, altText }) {
  return (
    <div className="avatar-container">
      <img className="avatar-image" src={imageSrc} alt={altText} />
    </div>
  );
}

Avatar.propTypes = {
  imageSrc: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
};

export default Avatar;
