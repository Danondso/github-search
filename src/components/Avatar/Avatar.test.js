import React from 'react';
import { FaBeer } from 'react-icons/fa';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Avatar from './Avatar';

const onClickMock = jest.fn();
beforeEach(() => {
  onClickMock.mockReset();
});

const testImageUrl = 'https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.distractify.com%2Fbrand-img%2FEm7U9Ugzx%2F0x0%2Fscreen-shot-2020-06-18-at-74312-pm-1592527404184.png&f=1&nofb=1';

describe('Avatar Test Suite', () => {
  test('renders avatar with all required props', () => {
    render(
      <Avatar imageSrc={testImageUrl} altText="Test Image" />,
    );
    const image = screen.getByAltText('Test Image');
    expect(image).toBeInTheDocument();
  });
});
