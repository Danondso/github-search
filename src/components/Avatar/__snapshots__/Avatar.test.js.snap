// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Test Suite renders avatar with all required props 1`] = `
<img
  alt="Test Image"
  class="avatar-image"
  src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fmedia.distractify.com%2Fbrand-img%2FEm7U9Ugzx%2F0x0%2Fscreen-shot-2020-06-18-at-74312-pm-1592527404184.png&f=1&nofb=1"
/>
`;
